packages/flutter/lib/src/cupertino/theme.dart

```diff
@@ -0,0 +1,391 @@
+@immutable
+class CupertinoThemeData extends Diagnosticable {
+  /// Create a [CupertinoTheme] styling specification.
+  ///
+  /// Unspecified parameters can be derived from another parameter or default
+  /// to an iOS default style.
+  const CupertinoThemeData({
+    Brightness brightness,
+    Color primaryColor,
+    Color primaryContrastingColor,
+    CupertinoTextTheme textTheme,
+    Color barBackgroundColor,
+    Color scaffoldBackgroundColor,
+    Color tableBackgroundColor,
+  }) : this.raw(
+        brightness,
+        primaryColor,
+        primaryContrastingColor,
+        textTheme,
+        barBackgroundColor,
+        scaffoldBackgroundColor,
+        tableBackgroundColor,
+      );
+
+  /// A raw constructor used by subclasses to populate the superclass's fields
+  /// to get superclass behaviors not overriden by subclasses.
+  ///
+  /// Same as the default constructor but with positional arguments to avoid
+  /// forgetting any.
+  @protected
+  const CupertinoThemeData.raw(
+    this._brightness,
+    this._primaryColor,
+    this._primaryContrastingColor,
+    this._textTheme,
+    this._barBackgroundColor,
+    this._scaffoldBackgroundColor,
+    this._tableBackgroundColor,
+  );
+
+  bool get _isLight => brightness == Brightness.light;
+
+  final Brightness _brightness;
+  /// The general brightness theme of the [CupertinoThemeData].
+  ///
+  /// Affects all other theme properties when unspecified. Defaults to
+  /// [Brightness.light].
+  ///
+  /// If coming from a Material [Theme] and unspecified, [brightness] will be
+  /// derived from the Material [ThemeData]'s `brightness`.
+  ///
+  /// Reading this property from [CupertinoTheme.of] will create a dependency
+  /// from the [BuildContext] to changes in this property.
+  Brightness get brightness => _brightness ?? Brightness.light;
+
+  final Color _primaryColor;
+  /// A color used on interactive elements of the theme.
+  ///
+  /// This color is generally used on text and icons in buttons and tappable
+  /// elements. Defaults to blue or orange when [brightness] is light or dark.
+  ///
+  /// If coming from a Material [Theme] and unspecified, [primaryColor] will be
+  /// derived from the Material [ThemeData]'s `colorScheme.primary`. However, in
+  /// iOS styling, the [primaryColor] is more sparsely used than in Material
+  /// Design where the [primaryColor] can appear on non-interactive surfaces like
+  /// the [AppBar] background, [TextField] borders etc.
+  ///
+  /// Reading this property from [CupertinoTheme.of] will create a dependency
+  /// from the [BuildContext] to changes in this property.
+  Color get primaryColor {
+    return _primaryColor ??
+        (_isLight ? CupertinoColors.activeBlue : CupertinoColors.activeOrange);
+  }
+
+  final Color _primaryContrastingColor;
+  /// A color used on content that contrast against a [primaryColor] background.
+  ///
+  /// This color is used on text and icons against a filled [CupertinoButton]
```

Comment:
For example, this color is used for a [CupertinoButton]'s text and icons when the button's background is [primaryColor].

Reply:
done